api/src/main/java/org/openmrs/module/fhir/api/impl/base/BaseConditionImpl.java

```
@@ -0,0 +1,74 @@
+/*
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.fhir.api.impl.base;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.hl7.fhir.dstu3.model.Condition;
+import org.openmrs.Obs;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.fhir.api.util.FHIRConditionUtil;
+import org.openmrs.module.fhir.api.util.FHIRUtils;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BaseConditionImpl {
+
+	/**
+	 * @see org.openmrs.module.fhir.api.helper.ConditionHelper#getCondition(String)
+	 */
+	public Condition getCondition(String uuid) {
+		int[] conceptsAsConditions = FHIRUtils.getConceptIdsOfConditions();
+		Obs obs = Context.getObsService().getObsByUuid(uuid);
+		if (obs == null || obs.isVoided() || conceptsAsConditions == null || !ArrayUtils.contains
+				(conceptsAsConditions, obs.getConcept().getId())) {
+			return null;
+		}
+
+		return FHIRConditionUtil.generateFHIRConditionFromOpenMRSObs(obs);
+	}
+
+	/**
+	 * @see org.openmrs.module.fhir.api.helper.ConditionHelper#createCondition(Condition)
+	 */
+
+	public Condition createCondition(Condition condition) {
+		List<String> errors = new ArrayList<>();
+		if (condition.isEmpty()) {
+			errors.add("Condition resource is null");
```

Can we throw an exception instead of simply returning null?